src/librbd/io/QosImageDispatch.cc

```diff
@@ -302,12 +307,15 @@ bool QosImageDispatch<I>::needs_throttle(
                       Tag{image_dispatch_flags, on_dispatched}, flag)) {
       ldout(cct, 15) << "on_dispatched=" << on_dispatched << ", "
                      << "flag=" << flag << dendl;
-      all_qos_flags_set = false;
     } else {
-      all_qos_flags_set = set_throttle_flag(image_dispatch_flags, flag);
+      flags_to_set |= flag;
     }
   }
-  return !all_qos_flags_set;
+
+  // flags_to_set should never be zero because a single op cannot activate all throttles
```

Comment:
The line was too long, instead of just wrapping it suggesting a small clarification:
```suggestion
  // flags_to_set should never be zero because a single op cannot
  // activate all throttles (and m_throttles can't be empty)
```